### ISSUE TITLE
All Telemetry Must Be Optional - Remove Tracking Pixel & Updates API

### ISSUE THREAD

**Issue by codyogden** [umami-software/umami#1369]
In #1173, I noted that this project was using the logo in a way to presumably gather metrics on version usage (and who knows what else with full requests going from clients to a third-party hosted solution). I suggested that we remove **all** external calls from this project and make any telemetry completely optional. The logo was removed as an external resource (thank you!), and replaced with a seemingly non-optional tracking pixel.

URL of tracking pixel: https://i.umami.is/a.png?v=1.36.1

URL of Updates API: https://api.umami.is/v1/updates?v=1.36.1

**This project cannot continue to advertise itself as "privacy-focused" and also track its own users without their consent.**

Telemetry with open source projects like this is perfectly reasonable expectation, but it needs to be optional. Also it should not originate from the user's browser. Specific features of this project expose public views of analytics data which means you're potentially capturing non-consenting user's information. A better way to track usage is to log it (optionally) during the build time.

**Comment by mikecao** [comment]
Hi Cody, I totally understand your concern. We only capture the version information to get a better idea of what versions people are using so we can provide better support. But of course you would have to trust that we are only doing that, so I get that that would be a concern.

If you look at the code, you'll note that both can be disabled via `DISABLE_UPDATES` and `DISABLE_TELEMETRY` environment variables. The code for the pixel is here:
https://github.com/umami-software/umami/blob/f1717ce5334c3c1655cbf4fe35ca6c3d751d32d8/components/layout/Footer.js#L31

and for the updates here:
https://github.com/umami-software/umami/blob/f1717ce5334c3c1655cbf4fe35ca6c3d751d32d8/components/layout/Header.js#L22

The updates api is not really used for tracking. It's to future proof our update process so we can send a custom payload back, such as custom messages and urls to the client depending on their version. For instance some future updates will require major updates to the database and we want to warn the user ahead of time to prevent any data loss.

So with the ability to disable all external calls via the environment variables currently, is that enough to satisfy your concerns? We welcome and value your input.

**Comment by codyogden** [comment]
On `v1.36.1` using the docker dist to deploy, I have the following (old & new) env vars set to disable telemetry, but I'm still seeing the tracking pixel called on the front end. https://analytics.kbg.rip

![Screen Shot 2022-07-29 at 8 43 04 PM](https://user-images.githubusercontent.com/11383626/181865463-1734c2c2-044f-4d3d-ba2c-523be7da5c20.png)

![Screen Shot 2022-07-29 at 8 43 49 PM](https://user-images.githubusercontent.com/11383626/181865482-9ebe6fcb-9b1a-41c2-a958-f4c586f9fecf.png)

**Comment by mikecao** [comment]
Yes, because of the way Next.js defines some variables at build time versus run time, you have to build your own docker image for certain variables. See https://umami.is/docs/environment-variables.

We are actively trying to come up with a solution. See similar situation in #1354. We may just have to force docker to run build at startup directly from the source so that all variables are treated equally. Most people who pull from the repo directly have to run build anyways so they aren't affected. This issue is specifically for docker, sorry for the confusion. We will try to remedy this as soon as possible.

**Comment by codyogden** [comment]
I'd suggest shipping telemetry separately from any component-logic. `next/server` is already running, so we can SSR the dashboard pages (meh), or ship a small telemetry script that either injects the pixel image or does nothing based on the current environment.

Rewrite `/telemetry.js` (e.g.) to `/api/scripts/telemetry` so you can take advantage of cache headers, etc but still have logic that affects whether telemetry runs.

```typescript
export default function handler(req: NextApiRequest, res: NextApiResponse) {
    res.setHeader('content-type', 'text/javascript');
    res.send(`console.log('${(process.env.DISABLE_TELEMETRY) ? 'tracking disabled' : 'tracking enabled'}')`);
}
```

Use `next/script` to run the telemetry script:

```jsx
<Script src='/telemetry.js' />
```

**Comment by mikecao** [comment]
Implemented in a4a68882483478116d8cef9b7f65c31216704847. Will go out with the next release. Thanks for the help @codyogden!

**Comment by mikecao** [comment]
Fixed in v1.37.0.
